src/main/java/org/apache/commons/imaging/formats/icns/IcnsDecoder.java

```diff
@@ -206,22 +206,24 @@ public static BufferedImage decodeImage(final IcnsImageParser.IcnsElement[] icns
             throws ImageReadException {
         final IcnsImageParser.IcnsElement imageElement = icnsElements[index];
         final IcnsType imageType = IcnsType.findImageType(imageElement.type);
+
+        //refactored in order to use contains instead of 11 conditions to reduce Cyclomatic complexity
```

Comment:
I don't think the refactoring should be in this PR.

Reply:
Fixed it! :) plz review again (branch on top of a branch which made the problem)